thingsboard_gateway/connectors/modbus_async/slave.py

```diff
@@ -0,0 +1,233 @@
+    def __init__(self, **config):
+        super().__init__()
+        self.daemon = True
+        self.stopped = False
+        self._log = config['logger']
+        self.connector = config['connector']
+        self.name = "Modbus slave processor for unit " + str(config['unitId']) + " on host " + str(
+            config['host']) + ":" + str(config['port'])
+
+        self.callback = config['callback']
+
+        self.unit_id = config['unitId']
+        self.host = config['host']
+        self.port = config['port']
+        self.type = config.get('type', 'tcp').lower()
+        self.method = config['method']
+        self.tls = config.get('tls', {})
+        self.timeout = config.get('timeout')
+        self.retry_on_empty = config.get('retryOnEmpty', False)
+        self.retry_on_invalid = config.get('retryOnInvalid', False)
+        self.retries = config.get('retries', 3)
+        self.baudrate = config.get('baudrate', 19200)
+        self.stopbits = config.get('stopbits', Defaults.Stopbits)
+        self.bytesize = config.get('bytesize', Defaults.Bytesize)
+        self.parity = config.get('parity', Defaults.Parity)
+        self.strict = config.get('strict', Defaults.Strict)
+        self.repack = config.get('repack', False)
+        self.word_order = config.get('wordOrder', 'LITTLE').upper()
+        self.byte_order = config.get('byteOrder', 'LITTLE').upper()
+
+        self.attributes_updates_config = config.get('attributeUpdates', [])
+        self.rpc_requests_config = config.get('rpc', [])
+
+        self.connect_attempt_time_ms = self.set_connect_attempt_time_ms(config.get('connectAttemptTimeMs', 500))
+        self.wait_after_failed_attempts_ms = self.set_wait_after_failed_attempts_ms(
+            config.get('waitAfterFailedAttemptsMs', 300000))
+        self.connection_attempt = self.set_connection_attempt(config.get('connectionAttempt', 5))
+
+        self.device_name = config['deviceName']
+        self.device_type = config.get('deviceType', 'default')
+
+        self.poll_period = config['pollPeriod'] / 1000
+
+        self.last_connect_time = 0
+        self.last_polled_time = 0
+        self.last_connection_attempt_time = 0
+        self.connection_attempt_count = 0
+
+        self.downlink_converter = self.__load_downlink_converter(config)
+
+        self.uplink_converter_config = BytesUplinkConverterConfig(**config)
+        self.uplink_converter = self.__load_uplink_converter(config)
+
+        self.master = None
+        self.available_functions = None
+
+        self.start()
+
+    def __timer(self):
+        self.__send_callback()
+
+        while not self.stopped and not self.connector.is_stopped():
+            if monotonic() - self.last_polled_time >= self.poll_period:
+                self.__send_callback()
+
+            sleep(.001)
+
+    # TODO: refactor it to use asyncio Task
+    def __send_callback(self):
+        try:
+            self.callback(self)
+            self.last_polled_time = monotonic()
```

Comment:
please move `self.last_polled_time = monotonic()` before try, to avoid spamming in logs in case of failing.